client/customloaders/customfileloader/index.js

```diff
@@ -0,0 +1,59 @@
+var Promise = require('bluebird');
+var fse = Promise.promisifyAll(require('fs-extra'));
+var loaderUtils = require('loader-utils');
+var mime = require('mime');
+var path = require('path');
+
+module.exports = function (content) {
+  this.cacheable && this.cacheable();
+  if (!this.emitFile) {
+    throw new Error('emitFile is required from module system');
+  }
+  var query = loaderUtils.parseQuery(this.query);
+  var url = loaderUtils.interpolateName(this, query.name || '[hash].[ext]', {
```

Comment:
Is the `'[hash].[ext]'` some naming scheme for our files or some Webpack output? I don't really understand what's happening here.